svc-task/thandle/thandle.go

```diff
@@ -125,10 +125,13 @@ func (ts *TasksRPC) deleteCompletedTask(taskID string) error {
 		log.Printf("error while deleting the main task: %v", err)
 		return err
 	}
-	err = ts.DeleteTaskIndex(taskID)
-	if err != nil {
-		log.Printf("error while deleting the main task: %v", err)
-		return err
+	//CompletedTaskIndex has only completed task which has subtasks associated with them. So delete index only when condition is met.
```

Comment:
what kind of errors may happen here ? Other than already having deleted task are there any other scenario we need to mind ?

Reply:
We are checking if any error is returned while deleting data from DB . This code is still there but in if condition below in L:132